src/Network/Kontiki/Raft.hs

```diff
@@ -38,7 +38,7 @@ import qualified Network.Kontiki.Raft.Leader as Leader
 -- and `state' of the node, runs the Raft protocol and  
 -- returns the new state of the node and a list of commands that 
 -- should be executed by the driver.
-handle :: (Functor m, Monad m, MonadLog m a) 
+handle :: (Monad m, MonadLog m a) 
```

Comment:
This removes compatibility with older compilers. The 'Redundant Constraints' warning of GHC8.0.1 was basically a mistake and is (for good reason) being removed from `-Werror` again, if I'm not mistaken (check GHC Trac). Would it be possible to add `-fno-warn-redundant-constraints` to the GHC options in the Cabal file instead (under a version check)?